worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -637,6 +639,12 @@ private void initializeFiles() throws IOException {
         hdfsSortedOutput =
             StorageManager.hadoopFs().create(new Path(sortedFilePath), true, 256 * 1024);
       } else {
+        if (memCacheManager.contains(originFilePath)) {
+          FileChannel channel = FileChannelUtils.createWritableFileChannel(originFilePath);
+          channel.write(memCacheManager.getCache(originFilePath).nioBuffer());
```

Comment:
The shuffle file is in memory. I think you can sort the file in memory and return the sorted shuffle data.

Reply:
IMO supporting sort in memory cache is out of this PR's scope, I think this is ok to fallback to disk-based when sort happens.